source/slang-capture-replay/slang-global-session.cpp

```diff
@@ -28,11 +28,24 @@ namespace SlangCapture
         m_actualGlobalSession->release();
     }
 
-    ISlangUnknown* GlobalSessionCapture::getInterface(const Guid& guid)
+    SLANG_NO_THROW SlangResult SLANG_MCALL GlobalSessionCapture::queryInterface(SlangUUID const& uuid, void** outObject) 
     {
-        if(guid == ISlangUnknown::getTypeGuid() || guid == IGlobalSession::getTypeGuid())
-            return asExternal(this);
-        return nullptr;
+        if (uuid == ISlangInternal::getTypeGuid())
+        {
+            // Special case to cast directly into internal type
+            // NOTE! No addref(!)
+            this->m_actualGlobalSession->queryInterface(uuid, outObject);
```

Comment:
I think it should be fine to have a addref call here so that we use a consistent convention throughout the codebase.

Reply:
When I had an addRef (and a release) it keepscrashing with huge memory leak dumps. 
I'm not quite sure what was going on, then I found the same workaround in `EndToEndCompileRequest::queryInterface` here (https://github.dev/shader-slang/slang/blob/f545ef30052de7b4cab389cb58ffb0ea67ffb32c/source/slang/slang.cpp#L3040) and I assumed that using ref-counting during a simple cast was causing problems with the system.

Reply:
Okay, I think the problem is that we use `asInternal` in a lot of places and we get a `Session*` as a raw pointer. None of the places take care of releasing the pointer explicitly, because thus far, Session* was a simple cast with no ref-counting

Updating `asInternal` to use `ComPtr` fixes this issue